python/lsst/ts/salobj/sal_info.py

```diff
@@ -227,19 +249,17 @@ def __init__(self, domain, name, index=0):
         # Create the publisher and subscriber. Both depend on the DDS
         # partition, and so are created here instead of in Domain,
         # where most similar objects are created.
-        self.partition_name = os.environ.get("LSST_DDS_DOMAIN")
-        if self.partition_name is None:
-            raise RuntimeError("Environment variable $LSST_DDS_DOMAIN not defined")
-
-        partition_qos_policy = dds.PartitionQosPolicy([self.partition_name])
-
-        publisher_qos = domain.qos_provider.get_publisher_qos()
-        publisher_qos.set_policies([partition_qos_policy])
-        self.publisher = domain.participant.create_publisher(publisher_qos)
-
-        subscriber_qos = domain.qos_provider.get_subscriber_qos()
-        subscriber_qos.set_policies([partition_qos_policy])
-        self.subscriber = domain.participant.create_subscriber(subscriber_qos)
+        self.partition_prefix = os.environ.get("LSST_DDS_PARTITION_PREFIX")
+        if self.partition_prefix is None:
```

Comment:
I think it would be good to add a warning if `LSST_DDS_DOMAIN` is set even if `LSST_DDS_PARTITION_PREFIX` is also set. Just to make sure people is aware of what is happening.

Reply:
How about if I warn only if they are different?